docs/source/Mapping.rst

```diff
@@ -0,0 +1,152 @@
+
+    .. code-block:: console
+
+                      ┌───┐                      ░ ┌───┐ ░                       ░ ┌─┐
+	  q_0 -> q_0: ┤ H ├──■───X───────────────░─┤ T ├─░───────────────X───■───░─┤M├─────────
+	              └───┘┌─┴─┐ │               ░ ├───┤ ░               │ ┌─┴─┐ ░ └╥┘┌─┐
+	  q_1 -> q_1: ─────┤ X ├─X───■───X───────░─┤ T ├─░───────X───■───X─┤ X ├─░──╫─┤M├──────
+	                   └───┘   ┌─┴─┐ │       ░ ├───┤ ░       │ ┌─┴─┐   └───┘ ░  ║ └╥┘┌─┐
+	  q_2 -> q_2: ─────────────┤ X ├─X───■───░─┤ T ├─░───■───X─┤ X ├─────────░──╫──╫─┤M├───
+	                           └───┘   ┌─┴─┐ ░ ├───┤ ░ ┌─┴─┐   └───┘         ░  ║  ║ └╥┘┌─┐
+	  q_3 -> q_3: ─────────────────────┤ X ├─░─┤ T ├─░─┤ X ├─────────────────░──╫──╫──╫─┤M├
+	                                   └───┘ ░ └───┘ ░ └───┘                 ░  ║  ║  ║ └╥┘
+	      meas: 4/══════════════════════════════════════════════════════════════╩══╩══╩══╩═
+                                                                                    0  1  2  3
+
+This mapping uses 4 SWAPs. As every SWAP decomposes into 3 CNOT gates, this is a very costly overhead for such a small circuit.
+Minimizing the number of additional gates introduced by mapping a quantum circuit is precisely the job of *QMAP*.
+
+Exact Mapping
+#############
+
+The *exact mapper* implemented in *QMAP* maps quantum circuits using the *minimal* number of SWAP gates. For directional architectures it furthermore tries to minimize the number of additional Hadamard gates. Using the exact mapper is as simple as:
+
+    .. code-block:: python3
+
+        qc_mapped, res = qmap.compile(qc, arch, method="exact")
+
+        print(qc_mapped.draw(fold=-1))
+
+        print("Additional gates: %d" % res.json()["statistics"]["additional_gates"])
+	print("Additional SWAPs: %d" % res.json()['statistics']['swaps'])
+        print("Runtime:          %f" % res.json()['statistics']['mapping_time'])
+
+    .. code-block:: console
+
+		                     ┌───┐┌───┐┌───┐     ┌─┐
+	q_3 -> 0 ────────────────────┤ X ├┤ T ├┤ X ├─────┤M├───────────────────
+	                        ┌───┐└─┬─┘├───┤└─┬─┘┌───┐└╥┘          ┌─┐
+	q_2 -> 1 ────────────■──┤ X ├──■──┤ T ├──■──┤ X ├─╫───■───────┤M├──────
+	         ┌───┐     ┌─┴─┐└─┬─┘┌───┐└───┘     └─┬─┘ ║ ┌─┴─┐     └╥┘┌─┐
+	q_0 -> 2 ┤ H ├──■──┤ X ├──■──┤ T ├────────────■───╫─┤ X ├──■───╫─┤M├───
+	         └───┘┌─┴─┐├───┤     └───┘                ║ └───┘┌─┴─┐ ║ └╥┘┌─┐
+	q_1 -> 3 ─────┤ X ├┤ T ├──────────────────────────╫──────┤ X ├─╫──╫─┤M├
+	              └───┘└───┘                          ║      └───┘ ║  ║ └╥┘
+	    c: 4/═════════════════════════════════════════╩════════════╩══╩══╩═
+	                                                  3            2  0  1
+	Additional gates: 2
+	Additional SWAPs: 2
+        Runtime:          0.015844
+
+Not only does this solution yield less SWAPs, by default the :code:`compile` method optimizes SWAP gates in a post-processing step.
+
```

Comment:
This could probably use a subsection to talk about search space limitations and refer to the corresponding paper. If you feel comfortable writing that, go for it. If not, let's create an issue for that and tag Sarah in it, as she is the author of that paper.

Reply:
Completely agree. Didn't do it on purpose because I don't feel comfortable writing about it. Gonna make it an issue.

Reply:
Sounds good. I'll resolve once the issues are created.